### PR TITLE
Introduce a parameter pack for function 'daml2ts'

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Use a parameter pack for `daml2ts` as per https://github.com/digital-asset/daml/pull/4799#discussion_r387883596

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -130,18 +130,28 @@ packageNameText pkgId mbPkgIdent = maybe (unPackageId pkgId) unPackageName mbPkg
...
 
 -- Gives the scope 'foo' given a directory path like '/path/to/foo'.
 -- The scope 'foo' is the '@foo' part of a package name in an import
 -- declaration.
 scopeOfScopeDir :: FilePath -> Scope
 scopeOfScopeDir = Scope . takeFileName
 
+data Daml2TsParams = Daml2TsParams
+  { opts :: Options  -- cli args
+  , pm :: Map.Map PackageId (Maybe PackageName, Package)
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: maybe just call this `packageMap`? `pm` seems a bit too generic and having worked with LLVM, I always things of `PassManager`.

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 2]
```diff
@@ -130,18 +130,28 @@ packageNameText pkgId mbPkgIdent = maybe (unPackageId pkgId) unPackageName mbPkg
 
 newtype Scope = Scope {unscope :: String}
 newtype Dependency = Dependency {undependency :: T.Text}  deriving (Eq, Ord)
+newtype SdkVersion = SdkVersion String
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: No need to introduce your own newtype, you can use `DA.Daml.Project.Types.SdkVersion`.

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Ok!

File: language-support/ts/codegen/src/TsCodeGenMain.hs (hunk 1 above)
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Going with `pkgMap` (consistency with other parameters).

**Comment by shayne-fletcher** [comment]
@cocreature please let me know what you think of this rewrite

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 3]
```diff
@@ -109,19 +111,22 @@ mergePackageMap ps = foldM merge Map.empty ps
 main :: IO ()
 main = do
     opts@Options{..} <- execParser optionsParserInfo
-    sdkVersion <- fromMaybe "0.0.0" <$> getSdkVersionMaybe
-    ps <- readPackages optInputDars
-    case mergePackageMap ps of
+    version <- DATypes.parseVersion . T.pack . fromMaybe "0.0.0" <$> getSdkVersionMaybe
```
[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: https://github.com/digital-asset/daml/pull/4861

**Pull request merged**